### PR TITLE
Refactored To Lastest SDK.

### PR DESCRIPTION
**Pull request by DongyiYang**
Working registration, discovery, target addition. An example config to start Kubeturbo is as follows:
```json
{
	"communicationConfig": {
		"serverMeta": {
			"turboServer": "https://1.2.3.4:8080"
		},
		"restAPIConfig": {
			"opsManagerUserName": "administrator",
			"opsManagerPassword": "password"
		}
	},
	"targetConfig": {
		"probeCategory":"CloudNative",
		"targetType":"Kubernetes",
		"address":"http://127.0.0.1:8080",
		"username":"kuberentes_target_user",
		"password":"kubernetes_target_password"
	}
}
```

### CODE REVIEW

File: pkg/discovery/probe/service_probe.go [hunk 1]
```diff
@@ -178,15 +180,15 @@ func (this *ServiceProbe) findBackendPodPerService(service *api.Service, endpoin
 	return serviceEndpointMap
 }
 
-func (this *ServiceProbe) buildEntityDTO(serviceName string, commoditiesBoughtMap map[*common.ProviderDTO][]*proto.CommodityDTO) (*proto.EntityDTO, error) {
+func (this *ServiceProbe) buildEntityDTO(serviceName string, commoditiesBoughtMap map[*builder.ProviderDTO][]*proto.CommodityDTO) (*proto.EntityDTO, error) {
```
Comment (enlinxu) [inline pkg/discovery/probe/service_probe.go]: This is only for VApp? Let's change the method name.

File: cmd/kubeturbo/app/kubeturbo_builder.go [hunk 2]
```diff
@@ -137,44 +117,35 @@ func (s *VMTServer) Run(_ []string) error {
...
-	kubeconfig, err := clientcmd.BuildConfigFromFlags(s.Master, s.Kubeconfig)
+	kubeConfig, err := clientcmd.BuildConfigFromFlags(s.Master, s.Kubeconfig)
 	if err != nil {
 		glog.Errorf("Error getting kubeconfig:  %s", err)
 		return err
 	}
 	// This specifies the number and the max number of query per second to the api server.
-	kubeconfig.QPS = 20.0
-	kubeconfig.Burst = 30
+	kubeConfig.QPS = 20.0
```
Comment (DongyiYang) [inline cmd/kubeturbo/app/kubeturbo_builder.go]: **QPS** indicates the maximum QPS to the master from this client.
Maximum **burst** for throttle.
You can find more[here](https://github.com/kubernetes/client-go/blob/master/rest/config.go#L103)

File: cmd/kubeturbo/app/kubeturbo_builder.go [hunk 3]
```diff
@@ -137,44 +117,35 @@ func (s *VMTServer) Run(_ []string) error {
...
+		glog.Errorf("Failed to generate correct TAP config: %s", err)
 		os.Exit(1)
 	}
 
-	glog.V(3).Infof("Finished creating turbo configuration: %++v", vmtMeta)
-
-	etcdclientBuilder := etcdhelper.NewEtcdClientBuilder().ServerList(s.EtcdServerList).SetTransport(s.EtcdCA, s.EtcdClientCertificate, s.EtcdClientKey)
-	etcdClient, err := etcdclientBuilder.CreateAndTest()
+	etcdClientBuilder := etcdhelper.NewEtcdClientBuilder().ServerList(s.EtcdServerList).SetTransport(s.EtcdCA, s.EtcdClientCertificate, s.EtcdClientKey)
+	etcdClient, err := etcdClientBuilder.CreateAndTest()
```
Comment (DongyiYang) [inline cmd/kubeturbo/app/kubeturbo_builder.go]: This is to create the etcdClient and then test if the etcdClient can reach the etcd cluster.

File: pkg/discovery/k8s_discovery_client.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+	nodeEntityDtos, err := kubeProbe.ParseNode()
+	if err != nil {
+		// TODO, should here still send out msg to server?
+		glog.Errorf("Error parsing nodes: %s. Will return.", err)
+		// TODO make error dto
+		return nil
+	}
+
+	podEntityDtos, err := kubeProbe.ParsePod(api.NamespaceAll)
```
Comment (DongyiYang) [inline pkg/discovery/k8s_discovery_client.go]: fixed

**Pull request merged**